libs/output-mapping/src/Configuration/Table/BaseConfiguration.php

```diff
@@ -0,0 +1,166 @@
+                ->booleanNode('incremental')->defaultValue(false)->end()
+                ->arrayNode('primary_key')
+                    ->prototype('scalar')
+                        // TODO: turn this on when all manifests will not produce array with an empty string
+                        // ->cannotBeEmpty()
+                    ->end()
+                ->end()
+                ->arrayNode('columns')
+                    ->prototype('scalar')
+                ->end()
+                ->end()
+                ->arrayNode('distribution_key')
+                    ->prototype('scalar')->cannotBeEmpty()->end()
+                ->end()
+                ->scalarNode('delete_where_column')->end()
+                ->arrayNode('delete_where_values')->prototype('scalar')->end()->end()
+                ->scalarNode('delete_where_operator')
+                    ->defaultValue('eq')
+                    ->beforeNormalization()
+                        ->ifInArray(['', null])
+                        ->then(function () {
+                            return 'eq';
+                        })
+                    ->end()
+                    ->validate()
+                    ->ifNotInArray(['eq', 'ne'])
+                        ->thenInvalid('Invalid operator in delete_where_operator %s.')
+                    ->end()
+                ->end()
+                ->scalarNode('delimiter')->defaultValue(self::DEFAULT_DELIMITER)->end()
+                ->scalarNode('enclosure')->defaultValue(self::DEFAULT_ENCLOSURE)->end()
+                ->arrayNode('metadata')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('key')->end()
+                            ->scalarNode('value')->end()
+                        ->end()
+                    ->end()
+                ->end()
+                ->arrayNode('column_metadata')
+                    ->useAttributeAsKey('name')
+                    ->prototype('array')
+                        ->prototype('array')
+                            ->children()
+                                ->scalarNode('key')->end()
+                                ->scalarNode('value')->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+                ->booleanNode('write_always')->defaultValue(false)->end()
+                ->arrayNode('tags')->prototype('scalar')->cannotBeEmpty()->end()->end()
+                ->scalarNode('manifest_type')->end()
+                ->booleanNode('has_header')->end()
+                ->scalarNode('description')->end()
+                ->variableNode('table_metadata')->end()
+                ->arrayNode('schema')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('name')->isRequired()->cannotBeEmpty()->end()
+                            ->arrayNode('data_type')->isRequired()
```

Comment:
data type není required (jen pokud má komponenta dataTypeSupport: authoritatitve)

Reply:
jo jasně, to tady nejde kontrolovat - vyhodím